test/src/test/java/org/corfudb/integration/LogReplicationDataReceiverIT.java

```diff
@@ -0,0 +1,305 @@
+            mapNameToMapStandby.put(mapName, tableStandby);
+        }
+
+
+        // Add Data for Snapshot Sync
+        writeToActive(0, NUM_WRITES);
+        // Confirm data does exist on Active Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapActive.values()) {
+            assertThat(map.count()).isEqualTo(NUM_WRITES);
+        }
+
+        // Confirm data does not exist on Standby Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapStandby.values()) {
+            assertThat(map.count()).isEqualTo(0);
+        }
+
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_ACTIVE;
+        startActiveLogReplicator();
+
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_STANDBY;
+        startStandbyLogReplicator();
+
+        verifyDataOnStandby(NUM_WRITES);
+
+        // Verify snapshot sync completed and cluster went into log entry sync
+        verifyInLogEntrySyncState();
+
+        stopActiveLogReplicator();
+        // Open table003 and table004 on active and standby with different metadata
+        Map<TableName, Table> standbyOnlyTables = new HashMap<>();
+        mapNameToMapStandby.clear();
+        mapNameToMapActive.clear();
+        for (int i = 3; i <= 4; i++) {
+            String mapName = TABLE_PREFIX + i;
+            TableName tableName = TableName.newBuilder()
+                    .setTableName(mapName)
+                    .setNamespace(NAMESPACE)
+                    .build();
+            Table<StringKey, IntValueTag, Metadata> tableStandby =
+                    corfuStoreStandby.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(NotFederatedTag.class, TableOptions.builder().build()));
+            mapNameToMapStandby.put(mapName, tableStandby);
+
+            Table<StringKey, IntValueTag, Metadata> tableActive =
+                    corfuStoreActive.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(IntValueTag.class, TableOptions.builder().build()));
+            mapNameToMapActive.put(mapName, tableActive);
+
+            standbyOnlyTables.put(tableName, tableStandby);
+        }
+        // Add Data for Snapshot Sync
+        writeToActive(0, NUM_WRITES);
+        startActiveLogReplicator();
+        verifyDataOnStandby(NUM_WRITES);
+
+        for (Map.Entry<TableName, Table> entry : standbyOnlyTables.entrySet()) {
+            CorfuRecord<TableDescriptors, TableMetadata> corfuRecord =
+                    standbyRuntime.getTableRegistry().getRegistryTable().get(entry.getKey());
+            Assert.assertFalse(corfuRecord.getMetadata().getTableOptions().getIsFederated());
+            Assert.assertTrue(corfuRecord.getMetadata().getTableOptions().getRequiresBackupSupport());
+        }
+    }
+
```

Comment:
can you also write a test where a stream/table has only been opened on the Source and verify that the registry table on the Sink gets updated with the new table's entry?

Reply:
Added a test for that, thanks!